src/main/java/org/breedinginsight/brapi/v2/ListController.java

```diff
@@ -0,0 +1,113 @@
+import org.breedinginsight.api.model.v1.response.DataResponse;
+import org.breedinginsight.api.model.v1.response.Response;
+import org.breedinginsight.api.model.v1.validators.QueryValid;
+import org.breedinginsight.brapi.v1.controller.BrapiVersion;
+import org.breedinginsight.brapi.v2.model.request.query.ListQuery;
+import org.breedinginsight.brapi.v2.services.BrAPIListService;
+import org.breedinginsight.brapps.importer.services.ExternalReferenceSource;
+import org.breedinginsight.model.Program;
+import org.breedinginsight.services.ProgramService;
+import org.breedinginsight.services.exceptions.DoesNotExistException;
+import org.breedinginsight.utilities.response.ResponseUtils;
+import org.breedinginsight.utilities.response.mappers.ListQueryMapper;
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+import java.util.List;
+import java.util.UUID;
+
+@Slf4j
+@Controller
+@Secured(SecurityRule.IS_AUTHENTICATED)
+public class ListController {
+
+    private final ProgramService programService;
+    private final BrAPIListService listService;
+    private final ListQueryMapper listQueryMapper;
+    private final String referenceSource;
+
+    @Inject
+    public ListController(ProgramService programService, BrAPIListService listService,
+                          ListQueryMapper listQueryMapper,
+                          @Property(name = "brapi.server.reference-source") String referenceSource) {
+        this.programService = programService;
+        this.listService = listService;
+        this.listQueryMapper = listQueryMapper;
+        this.referenceSource = referenceSource;
+    }
+
+    //@Get(BrapiVersion.BRAPI_V2 + "/lists")
+    @Get("/${micronaut.bi.api.version}/programs/{programId}" + BrapiVersion.BRAPI_V2 + "/lists{?queryParams*}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ProgramSecured(roleGroups = {ProgramSecuredRoleGroup.ALL})
+    public HttpResponse<Response<DataResponse<Object>>> getLists(
+            @PathVariable("programId") UUID programId,
+            @QueryValue @QueryValid(using = ListQueryMapper.class) @Valid ListQuery queryParams
+    ) throws DoesNotExistException, ApiException {
+        try {
+            Program program = programService
+                    .getById(programId)
+                    .orElseThrow(() -> new DoesNotExistException("Program does not exist"));
+
+            // get germplasm lists by default
+            BrAPIListTypes type = BrAPIListTypes.fromValue(queryParams.getListType());
+            if (type == null) {
+                type = BrAPIListTypes.GERMPLASM;
+            }
+            String source = queryParams.getExternalReferenceSource() == null ?
+                    String.format("%s/%s", referenceSource, ExternalReferenceSource.PROGRAMS.getName()) :
+                    queryParams.getExternalReferenceSource();
+            UUID id = queryParams.getExternalReferenceId() == null || queryParams.getExternalReferenceId().isBlank() ?
```

Comment:
I tried testing with only setting the `externalReferenceSource` query param and got no results back.  This line appears to be the reason why.  I don't think a default should be set for this (maybe not even for the externalReferenceSource either).  Changing this to allow nulls may require changes to downstream code in this call stack

Reply:
I removed the default settings for the xref params in `ListController`, added checks in `BrAPIListService` so that the a search request could be formed from the params, and added `BrAPIListDAO::getListsBySearch`. The xrefSource and XrefId params can now be set independently and return any lists. The service and DAO are set up to treat listType as an independent search param, too; however, the default value of germplasm is still being set in the controller unitl the existing germplasm code is updated. So, just setting xrefSource=`breeding-insight.net/dataset` will return no data because a listType=`germplasm` is also being used in the search.